packages/@o3r/rules-engine/src/engine/ruleset-executor.ts

```diff
@@ -267,7 +267,7 @@ export class RulesetExecutor {
               if (this.rulesEngine.debugMode) {
                 output.evaluation = handleRuleEvaluationDebug({ ...rule, inputFacts }, this.ruleset.name, output.actions, output.error, runtimeFactValues, factValues, oldFactValues);
               } else if (output.error) {
-                this.rulesEngine.logger?.error(output.error);
+                this.rulesEngine.logger?.error(output.error.toString());
```

Comment:
```suggestion
                this.rulesEngine.logger?.error(output.error instanceof Error ? output.error.toString() : typeof output.error === 'string' ? output.error : JSON.stringify(output.error));
```
as you can throw whatever you want 
when you catch you received an error of type `unknown`